media-converter/src/videoconv.rs

```diff
@@ -541,8 +541,11 @@ impl VideoConv {
 
         if self.sinkpad.peer_query(&mut query) {
             /* XXX: what? */
-            let res = query.get_result();
-            drop(res);
+            let _ = query.get_result();
+            /* Legend: This copies `res` and only then gets dropped ~ doing nothing 
+             * I'll just leave it commented, as it seems important to understanding 
+             * how the program works*/
+            //drop(res);
```

Comment:
I don't feel like I explained this well enough. 
It copies res, and then it drops that copy, leaving the original intact ~ only a unneeded overhead

Reply:
This one is weird. I don't remember the details, but there was some bizarre breakage if I didn't add the explicit drop() here. Hence my "what?" comment. I kind of suspect it was a compiler bug.

Reply:
Alright, admittedly I didn't test my changes. I trust the rust compiler a bit too much on that probably. I will either test them (what game do I even need to run to test this 🤔 ), or we just leave it there for safety.

Reply:
Taking another look at this one, we call `get_result` twice. In theory we can just drop all of the LOC before the match, right? E.g.:

```diff
@@ -540,9 +540,6 @@ impl VideoConv {
         let mut query = gst::query::Duration::new(gst::Format::Bytes);
 
         if self.sinkpad.peer_query(&mut query) {
-            /* XXX: what? */
-            let res = query.get_result();
-            drop(res);
             state.upstream_duration = match query.get_result() {
                 gst::format::GenericFormattedValue::Bytes(b) =>
                     *b,
```

And it should be functionally equivalent... right? I think we can try this and see if our QA hits a problem during testing.

Reply:
```rust
        unsafe {
            let mut fmt = mem::MaybeUninit::uninit();
            let mut pos = mem::MaybeUninit::uninit();

            gst_sys::gst_query_parse_duration(self.0.as_ptr(), fmt.as_mut_ptr(), pos.as_mut_ptr());

            GenericFormattedValue::new(from_glib(fmt.assume_init()), pos.assume_init())
        }
```
This is the code inside the get_results() function. If I'm honest, the function names don't look like they'd have side effects - so I reckon it should be alright removing those.